modules/renter/hostdb/hosttree/hosttree.go

```diff
@@ -0,0 +1,274 @@
+package hosttree
+
+import (
+	"crypto/rand"
+	"errors"
+	"sync"
+
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+var (
+	// ErrWeightTooHeavy is returned from a Fetch() call if a weight that exceeds
+	// the total weight of the tree is requested.
+	ErrWeightTooHeavy = errors.New("requested a too-heavy weight")
+
+	// ErrNegativeWeight is returned from an Insert() call if an entry with a
+	// negative weight is added to the tree. Entries must always have a positive
+	// weight.
+	ErrNegativeWeight = errors.New("cannot insert using a negative weight")
+
+	// ErrNilEntry is returned if a fetch call results in a nil tree entry. nodes
+	// should always have a non-nil entry, unless they have been Delete()ed.
+	ErrNilEntry = errors.New("node has a nil entry")
+
+	// ErrHostExists is returned if an Insert is called with a public key that
+	// already exists in the tree.
+	ErrHostExists = errors.New("host already exists in the tree")
+
+	// ErrNoSuchHost is returned if Remove is called with a public key that does
+	// not exist in the tree.
+	ErrNoSuchHost = errors.New("no host with specified public key")
```

Comment:
You would think, but in practice it's just clutter.

Unless you find yourself thinking "man, it'd be really nice if the exact error were exposed", there's no reason to export your errors. There are a few places in the code where we switch based on error (e.g. in the gateway I think we handle timeouts differently from other types of errors), but it's only a few and usually you are already going out of your way to get the two packages to communicate properly.